lib/DXIL/DxilOperations.cpp

```diff
@@ -3564,8 +3564,9 @@ void OP::UpdateCache(OpCodeClass opClass, Type *Ty, llvm::Function *F) {
 Function *OP::GetOpFunc(OpCode opCode, Type *pOverloadType) {
   if (opCode == OpCode::NumOpCodes)
     return nullptr;
-  if (!IsOverloadLegal(opCode, pOverloadType))
-    return nullptr;
+  DXASSERT(IsOverloadLegal(opCode, pOverloadType),
+           "otherwise the caller requested illegal operation overload (eg HLSL "
+           "function with unsupported types for mapped intrinsic function)");
```

Comment:
There is the question of whether to keep the assert here.

I think if we do not have a more general solution planned for the near future that doesn't rely on this code path, we should not have the assert, since we know we rely on violating this assert for perfectly legal code.

In my original change to add tests in #6437, I was going to remove the assert and add a comment to explain why:
https://github.com/microsoft/DirectXShaderCompiler/pull/6437/commits/f72d2e2702c7f058e8b755cddd04def2332edbc7#diff-e85a54111eb6509ee6bf740224386600db9c32fa2fc38d9b95cbb202f0882800L3570

I think this change makes sense for main now (removing the assert and adding the comment).

Reply:
And if we do remove the assert, we won't need the `UNSUPPORTED: asserts` in the tests either.

Reply:
Updated.